### ISSUE TITLE
Tuple slicing leads to wrong type

### ISSUE THREAD

**Issue by twoertwein** [microsoft/pyright#1886]
**To Reproduce**
```py
def test() -> tuple[int, str]:
    return 0, ""

number, string = test()[:2]
string.strip()
```
```
error: Cannot access member "strip" for type "int"
    Member "strip" is unknown (reportGeneralTypeIssues)
```

**Expected behavior**
string is of type `str`.

**Comment by erictraut** [comment]
Yes, pyright currently doesn't have any special-case knowledge of slicing for tuples. I think it's a reasonable request. Mypy handles this case.

**Comment by erictraut** [comment]
Thanks for the suggestion. I've implemented support for tuple slicing when the following cases hold:
1. The tuple has a well-known length (i.e. it's not a homogeneous tuple of indeterminate length)
2. The slice expression contains a start and/or end expression
3. The slice expression does not contain a span expression
4. The types of the start and/or end expressions can be determined statically to be integer literals
5. The start and end values (negative values are supported) are within bounds of the known tuple length

I think this covers all of the common cases.

Here's the change with unit tests: https://github.com/microsoft/pyright/commit/1b1aeb620c92b18024e2686e73c375bee6a4cbb3

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.142, which I just published. It will also be included in the next release of pylance.
